lib/backend/pgbk/common/gcp_config.go

```diff
@@ -0,0 +1,134 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package pgcommon
+
+import (
+	"net/url"
+	"slices"
+	"strings"
+
+	"cloud.google.com/go/cloudsqlconn"
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgx/v5"
+
+	apiutils "github.com/gravitational/teleport/api/utils"
+	gcputils "github.com/gravitational/teleport/lib/utils/gcp"
+)
+
+// gcpConfig defines GCP specific configs.
+type gcpConfig struct {
+	// connectionName is the GCP connection name in format of
+	// project:region:instance. The connection name is required by the
+	// connector libraries as the connection target.
+	connectionName string
+	// ipType specifies the type of IP used for GCP connection.
+	ipType gcpIPType
+	// serviceAccount is the target service account that will login the
+	// database.
+	serviceAccount string
+}
+
+func (c *gcpConfig) check() error {
+	if c.connectionName == "" {
+		return trace.NotFound("missing #%s (hint: project:region:instance)", gcpConnectionNameParam)
+	}
+	if err := c.ipType.check(); err != nil {
+		return trace.Wrap(err)
+	}
+	if err := gcputils.ValidateGCPServiceAccountName(c.serviceAccount); err != nil {
+		return trace.Wrap(err, "IAM database user for service account should have usernames in format of <service_account_name>@<project_id>.iam")
+	}
+	return nil
+}
+
+const (
+	gcpConnectionNameParam = "gcp_connection_name"
+	gcpIPTypeParam         = "gcp_ip_type"
+)
+
+func gcpConfigFromConnConfig(connConfig *pgx.ConnConfig) (*gcpConfig, error) {
+	// TODO(greedy52) maybe support DSN format?
```

Comment:
Let me update this, and share you the look after

Reply:
https://docs-6pnqngs1d-goteleport.vercel.app/docs/reference/backends/#google-cloud-iam-authentication

It looks like this now:
```
teleport:
  storage:
    type: postgresql
    auth_mode: gcp-cloudsql
    gcp_connection_name: project:location:instance
    conn_string: postgresql://service-account@project.iam@/teleport_backend
```
I was debating `gcp_connection_name/gcp_ip_type` vs `gcp.connection_name/ip_type`. With former, it more aligns with the parameter used for `pgevents`.

let me know how it looks.

Reply:
Thankfully we don't have prior art to deal with (azure auth always only uses the default credentials as configured via envvars) so I think we can go for `gcp_` rather than a sub object, so the parameters match exactly with the ones in `pgevents`.